src/components/grouped-character/grouped-character-test.stories.tsx

```diff
@@ -77,3 +78,30 @@ NewValidation.args = {
   separator: "-",
   ...getCommonTextboxArgs(),
 };
+
+export const GroupedCharacterComponent = ({
+  onChange,
+  ...props
+}: {
+  onChange?: (modifiedEvent: any) => void;
+}) => {
+  const [state, setState] = React.useState("");
+
+  const setValue = ({ target }: CustomEvent) => {
+    setState(target.value.rawValue);
+    if (onChange) {
+      onChange(target);
+    }
+  };
```

Comment:
This will throw a type error since the `onChange` function is expecting an event, not the element it is targeting.
```suggestion
  const setValue = (ev: CustomEvent) => {
    setState(ev.target.value.rawValue);
    if (onChange) {
      onChange(ev);
    }
  };
```